microformats_test.go

```diff
@@ -561,3 +561,20 @@ func Test_GetFirstPropValue(t *testing.T) {
 		}
 	}
 }
+
+func Test_ParseNodeNil(t *testing.T) {
+	tests := map[string]string{
+		"absolute": "<html><head><base href=\"https://example.com\"></head></html>",
+		"relative": "<html><head><base href=\"./something\"></head></html>",
+		"none":     "<html><head></head></html>", // parseNode(tt) == nil
+	}
+	for name, tt := range tests {
+		t.Run(name, func(t *testing.T) {
+			n, err := parseNode(tt)
+			if err != nil {
+				t.Fatalf("Error parsing HTML: %v", err)
+			}
+			ParseNode(n, nil)
```

Comment:
It's not obvious why we are calling ParseNode here, but not doing anything with the return value.  Maybe add a comment `// ParseNode shouldn't panic` ?